File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prod=1,l=nums.length,zc=0;
        int res[] = new int[l];


        for(int i =0;i<l;i++){
            if(nums[i]!=0){
            prod*=nums[i];
            }
            else
            zc++;
        }
        
        for(int i=0;i<l;i++){
            if(zc==0)
                res[i]=prod/nums[i];
                else{
                    if(zc>1)
                    prod=0;

                    if(nums[i]==0)
                    res[i]=prod;
                    else
                    res[i]=0;
                }
        }
        return res;
    }
}  

File: Data Structures & Algorithms/target-sum/submission-2.cpp
#include <bits/stdc++.h>
class Solution {
    vector<int> nums;
    int target;
    int res=0;
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        this->nums=nums;
        this->target=target;
        backTrack(0,0);
        return res;
    }

    void backTrack(int index,int sum){
        if(index==nums.size()){
            if(sum==target)
            res++;
        return; 
        }
        else if(index<nums.size()){
            backTrack(index+1,sum+nums[index]);
            backTrack(index+1,sum-nums[index]);
        }
    }
};
